js/analyzer.js

```diff
@@ -1,65 +1,232 @@
 /**
- * Created by danny on 3/19/15.
+ * Text Analyzer, accepts args for config and
+ * @param args
+ * @constructor
  */
+Analyzer = function (args){
+    this.config = args;
+    this.init();
+};
 
-/*text preprocessor*/
+/**
+ * Analyzer initialisation. Loads defaults and overloads custom settings.
+ */
+Analyzer.prototype.init = function() {
 
+    //init preprocessor if not exists
+    if (typeof yst_pp != 'object' || yst_pp.inputText != this.config.textString) {
+        yst_pp = new PreProcessor(this.config.textString);
+    }
+    //if no available function queues, make new queue
+    if (typeof this.queue != 'Array') {
+        this.queue = [];
+    }
+    //if custom queue available load queue, otherwise load default queue.
+    if(typeof this.config.queue != 'undefined' && this.config.queue.length != 0){
+        this.queue = this.config.queue;
+    }else{
+        this.queue = ['keywordDensity', 'subheaderChecker'];
+    }
+    //if no available keywords, load default array
+    if(typeof this.config.wordsToRemove == 'undefined'){
+        this.config.wordsToRemove =  [' a', ' in', ' an', ' on', ' for', ' the', ' and'];
+    }
+    //set default variables
+    this.keywordRegex = new RegExp(this.config.keyword);
+    this._output = [];
+}
+
+/**
+ * starts queue of functions executing the analyzer functions untill queue is empty.
+ */
+Analyzer.prototype.runQueue = function(){
+    //remove first function from queue and execute it.
+    if(this.queue.length > 0){
+        this._output.push(this[this.queue.shift()]());
+        this.runQueue();
+    }
+};
+/**
+ * clears current queue of functions, effectively stopping execution of the analyzer.
+ */
+Analyzer.prototype.abortQueue = function(){
+    //empty current Queue
+    this.queue = [];
+};
+
+
+/**
+ * checks the keyword density of given keyword against the cleantext stored in _store.
+ * @returns resultObject
+ */
+Analyzer.prototype.keywordDensity = function(){
+    if(yst_pp._store.wordcount >= 100) {
+        var keywordMatches = yst_pp._store.cleanText.match(new RegExp(this.config.keyword, 'g'));
+        if ( keywordMatches != null ) {
+            var keywordCount = keywordMatches.length;
+            var keywordDensity = (keywordCount / yst_pp._store.wordcount - (keywordCount - 1 * keywordCount)) * 100;
+        } else {
+            var keywordDensity = 0;
+        }
+        switch (true) {
```

Comment:
Please move the rating to a different function, so the functions have a single responsibility. In the case of `keywordDensity` I would expect a function that gets the keyword density. The keyword density rating would be a different function.